File: swift-bridge/Sources/WeatherKitBridge/SunEvents.swift
import Foundation
import WeatherKit

struct WKSunEventsPayload: Codable {
    let astronomicalDawn: String?
    let nauticalDawn: String?
    let civilDawn: String?
    let sunrise: String?
    let solarNoon: String?
    let sunset: String?
    let civilDusk: String?
    let nauticalDusk: String?
    let astronomicalDusk: String?
    let solarMidnight: String?
}

func wkPayload(from sunEvents: SunEvents) -> WKSunEventsPayload {
    WKSunEventsPayload(
        astronomicalDawn: sunEvents.astronomicalDawn.map(wkISO8601String),
        nauticalDawn: sunEvents.nauticalDawn.map(wkISO8601String),
        civilDawn: sunEvents.civilDawn.map(wkISO8601String),
        sunrise: sunEvents.sunrise.map(wkISO8601String),
        solarNoon: sunEvents.solarNoon.map(wkISO8601String),
        sunset: sunEvents.sunset.map(wkISO8601String),
        civilDusk: sunEvents.civilDusk.map(wkISO8601String),
        nauticalDusk: sunEvents.nauticalDusk.map(wkISO8601String),
        astronomicalDusk: sunEvents.astronomicalDusk.map(wkISO8601String),
        solarMidnight: sunEvents.solarMidnight.map(wkISO8601String)
    )
}

File: src/sun_events.rs
//! WeatherKit sun event types.

use serde::Deserialize;

/// Represents WeatherKit sun event values.
#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SunEvents {
    /// Matches the WeatherKit astronomical dawn value.
    pub astronomical_dawn: Option<String>,
    /// Matches the WeatherKit nautical dawn value.
    pub nautical_dawn: Option<String>,
    /// Matches the WeatherKit civil dawn value.
    pub civil_dawn: Option<String>,
    /// Matches the WeatherKit sunrise value.
    pub sunrise: Option<String>,
    /// Matches the WeatherKit solar noon value.
    pub solar_noon: Option<String>,
    /// Matches the WeatherKit sunset value.
    pub sunset: Option<String>,
    /// Matches the WeatherKit civil dusk value.
    pub civil_dusk: Option<String>,
    /// Matches the WeatherKit nautical dusk value.
    pub nautical_dusk: Option<String>,
    /// Matches the WeatherKit astronomical dusk value.
    pub astronomical_dusk: Option<String>,
    /// Matches the WeatherKit solar midnight value.
    pub solar_midnight: Option<String>,
}
